assets/app/components/site/Pages/pagesContainer.js

```diff
@@ -14,85 +16,65 @@ class Pages extends React.Component {
   }
 
   componentDidMount() {
-    siteActions.fetchFiles(this.props.site, this.getPath(this.props.params));
+    fetchFiles(this.props.site, this.props.params);
   }
 
   componentWillReceiveProps(nextProps) {
     const { params, site } = nextProps;
-    const nextFiles = this.getFilesByPath(site.files, this.getPath(params));
-    const files = this.getFilesByPath(site.files, this.getPath(params));
+    const nextFiles = getFilesByPath(site.files, getPath(params));
+    const files = getFilesByPath(site.files, getPath(params));
 
     if (files.length && nextFiles.length === files.length) return;
 
-    siteActions.fetchFiles(site, this.getPath(params));
-  }
-
-  getLinkFor(page, id, branch) {
-    const path = page.path;
-
-    return this.isDir(page) ?
-      `/sites/${id}/tree/${path}` : `/sites/${id}/edit/${branch}/${path}`;
+    fetchFiles(site, params);
   }
 
-  getPath(routeParams) {
-    const { splat, fileName } = routeParams;
-    let path = '';
+  render() {
+    const { site, params } = this.props;
 
-    if (splat) {
-      path = `${splat}/${fileName}`;
+    if (!site) {
+      return null;
     }
-    else if (fileName) {
-      path = fileName;
+
+    if (hasNavigationJsonContent(site)) {
+      return <NavigationJsonPageList site={ site } />;
     }
 
-    return path;
+    const path = getPath(params);
+    const files = getFilesByPath(site.files, path) || [];
+    return <PageList site={ site } files={ files }/>;
   }
+}
 
+const fetchFiles = (site, params) => {
+  siteActions.fetchFiles(site, getPath(params));
+};
 
-  getFilesByPath(files = [], startingPath = '') {
-    const isRoot = (startingPath === '');
-    const path = (!isRoot) ? `${startingPath}/` : '((?!/).)*$';
-    const startsWithPath = new RegExp(`^${path}`);
+const getPath = (routeParams) => {
```

Comment:
Do we ever want anyone outside of the pagesContainer object to call pagesContainer.getFilesByPath() or pagesContainer.getPath()? If we prevent people from calling such functions, then nobody will ever accidentally call these functions. The only way we have real "private" members in our javascript class is by hiding them from the outside.

I'm also actively trying to put our components on a diet with a goal of having them responsible only for rendering a particular state represented by properties. This is intended to start extracting such functionality. For instance, why does the view component responsible for extracting "splat" and "fileName" from a map and determining what to give siteActions.fetchFiles()? Would it be better to push that logic down to the siteActions since pagesContainer is the only caller of siteActions.fetchFiles(), especially when we're going to cut more getPath() calls in pagesContainer.componentWillReceiveProps()? Moving out methods like these puts more of a "what is that doing here?" spotlight on the code.